classes/DepositPackage.inc.php

```diff
@@ -598,68 +606,77 @@ public function updateDepositStatus() {
+				$this->_deposit->setStatus(PLN_PLUGIN_DEPOSIT_STATUS_PACKAGED | PLN_PLUGIN_DEPOSIT_STATUS_TRANSFERRED | PLN_PLUGIN_DEPOSIT_STATUS_RECEIVED | PLN_PLUGIN_DEPOSIT_STATUS_VALIDATED | PLN_PLUGIN_DEPOSIT_STATUS_SENT);
 				break;
 			default:
 				$this->_deposit->setExportDepositError('Unknown processing state ' . $processingState);
 				$this->_logMessage('Deposit ' . $this->_deposit->getId() . ' has unknown processing state ' . $processingState);
 				break;
 		}
 
+		// The deposit file can be dropped once it's received by the PKP PN
+		if ($this->_deposit->getReceivedStatus()) {
+			$this->remove();
+		} elseif (!file_exists($this->getAtomDocumentPath())) {
+			// Otherwise the package must still exist at this point, if it doesn't, we restart the deposit
+			$this->_deposit->setNewStatus();
+			$depositDao->updateObject($this->_deposit);
+			return;
+		}
+
+		// Handle error messages
+		if (in_array($processingState, ['hold', 'harvest-error', 'deposit-error', 'reserialize-error', 'virus-error', 'xml-error', 'payload-error', 'bag-error', 'status-error'])) {
+			$this->_deposit->setExportDepositError(__('plugins.generic.pln.status.error.' . $processingState));
+		}
+
 		$lockssState = $contentDOM->getElementsByTagName('category')->item(1)->getAttribute('term');
-		switch($lockssState) {
+		$this->_deposit->setLockssState($lockssState ?: null);
+		switch ($lockssState) {
 			case '':
 				// do nothing.
 				break;
-			// WARNING: The usage of 'received' is unknown and it may be removed in the future
-			case 'received':
-				$this->_deposit->setLockssReceivedStatus();
-				break;
-			// WARNING: The usage of 'syncing' is unknown and it may be removed in the future
-			case 'syncing':
 			case 'inProgress':
-				$this->_deposit->setLockssSyncingStatus();
+				$this->_deposit->setStatus(PLN_PLUGIN_DEPOSIT_STATUS_PACKAGED | PLN_PLUGIN_DEPOSIT_STATUS_TRANSFERRED | PLN_PLUGIN_DEPOSIT_STATUS_RECEIVED | PLN_PLUGIN_DEPOSIT_STATUS_VALIDATED | PLN_PLUGIN_DEPOSIT_STATUS_SENT | PLN_PLUGIN_DEPOSIT_STATUS_LOCKSS_RECEIVED);
 				break;
 			case 'agreement':
-				if(!$this->_deposit->getLockssAgreementStatus()) {
-					$this->remove();
-				}
-				$this->_deposit->setLockssAgreementStatus(true);
+				$this->_deposit->setStatus(PLN_PLUGIN_DEPOSIT_STATUS_PACKAGED | PLN_PLUGIN_DEPOSIT_STATUS_TRANSFERRED | PLN_PLUGIN_DEPOSIT_STATUS_RECEIVED | PLN_PLUGIN_DEPOSIT_STATUS_VALIDATED | PLN_PLUGIN_DEPOSIT_STATUS_SENT | PLN_PLUGIN_DEPOSIT_STATUS_LOCKSS_RECEIVED | PLN_PLUGIN_DEPOSIT_STATUS_LOCKSS_AGREEMENT);
+				$this->_deposit->setPreservedDate(Core::getCurrentDate());
 				break;
 			default:
 				$this->_deposit->setExportDepositError('Unknown LOCKSS state ' . $lockssState);
 				$this->_logMessage('Deposit ' . $this->_deposit->getId() . ' has unknown LOCKSS state ' . $lockssState);
 				break;
 		}
 
-		$this->_deposit->setLastStatusDate(time());
+		$this->_deposit->setLastStatusDate(Core::getCurrentDate());
```

Comment:
should this be part of a `setStatus` function?

Reply:
Yep! Everything should be done together.